server/core/src/objDesc.cpp

```diff
@@ -28,10 +29,84 @@
 
 int
 initL1desc() {
-    memset( L1desc, 0, sizeof( L1desc ) );
+    for (auto&& e : L1desc) {
+        init_l1desc(e);
+    }
+
     return 0;
 }
 
+auto init_l1desc(l1desc& _l1d) -> void
+{
+    _l1d.l3descInx = 0;
+    _l1d.inuseFlag = 0;
+    _l1d.oprType = 0;
+    _l1d.openType = 0;
+    _l1d.oprStatus = 0;
+    _l1d.dataObjInpReplFlag = 0;
+    _l1d.copiesNeeded = 0;
+    _l1d.bytesWritten = 0;
+    _l1d.dataSize = 0;
+    _l1d.replStatus = 0;
+    _l1d.chksumFlag = 0;
+    _l1d.srcL1descInx = 0;
+    _l1d.remoteL1descInx = 0;
+    _l1d.stageFlag = 0;
+    _l1d.purgeCacheFlag = 0;
+    _l1d.lockFd = 0;
+
+    _l1d.dataObjInp = nullptr;
+    _l1d.dataObjInfo = nullptr;
+    _l1d.otherDataObjInfo = nullptr;
+    _l1d.replDataObjInfo = nullptr;
+    _l1d.remoteZoneHost = nullptr;
+
+    _l1d.pluginData.clear();
+    _l1d.replica_token.clear();
+
+    // NOLINTBEGIN(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+    std::memset(_l1d.chksum, 0, sizeof(l1desc::chksum));
+    std::memset(_l1d.in_pdmo, 0, sizeof(l1desc::in_pdmo));
+    // NOLINTEND(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+} // init_l1desc
+
+auto copy_l1desc(l1desc& _dst, const l1desc& _src) -> void
+{
+    _dst.l3descInx = _src.l3descInx;
+    _dst.inuseFlag = _src.inuseFlag;
+    _dst.oprType = _src.oprType;
+    _dst.openType = _src.openType;
+    _dst.oprStatus = _src.oprStatus;
+    _dst.dataObjInpReplFlag = _src.dataObjInpReplFlag;
+    _dst.copiesNeeded = _src.copiesNeeded;
+    _dst.bytesWritten = _src.bytesWritten;
+    _dst.dataSize = _src.dataSize;
+    _dst.replStatus = _src.replStatus;
+    _dst.chksumFlag = _src.chksumFlag;
+    _dst.srcL1descInx = _src.srcL1descInx;
+    _dst.remoteL1descInx = _src.remoteL1descInx;
+    _dst.stageFlag = _src.stageFlag;
+    _dst.purgeCacheFlag = _src.purgeCacheFlag;
+    _dst.lockFd = _src.lockFd;
+
+    // TODO Should these be copied?
+    _dst.dataObjInp = _src.dataObjInp;
+    _dst.dataObjInfo = _src.dataObjInfo;
+    _dst.otherDataObjInfo = _src.otherDataObjInfo;
+    _dst.replDataObjInfo = _src.replDataObjInfo;
+    _dst.remoteZoneHost = _src.remoteZoneHost;
+
+    // TODO Should these be copied?
+    _dst.pluginData = _src.pluginData;
+    _dst.replica_token = _src.replica_token;
+
+    // NOLINTBEGIN(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+    // TODO Should these be copied?
+    std::memcpy(_dst.chksum, _src.chksum, sizeof(l1desc::chksum));
+    std::memcpy(_dst.in_pdmo, _src.in_pdmo, sizeof(l1desc::in_pdmo));
+    // NOLINTEND(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
```

Comment:
Should think about these.

Reply:
These should be copied because that is what the function is supposed to do and doing anything else feels more incorrect.

Reply:
A quick use of grep shows that the following member variables are not used:
- `otherDataObjInfo`
- `replDataObjInfo`
- `pluginData`

I'm going to ignore these variables and create an issue to deprecate them. They aren't used and their meaning is clear.

Reply:
Done.